packages/adp-tooling/test/unit/writer/index.test.ts

```diff
@@ -86,5 +86,146 @@ describe('ADP writer', () => {
                 )
             ).toMatchSnapshot();
         });
+
+        test('enable s4hana cloud', async () => {
+            const projectDir = join(outputDir, 's4hana');
+            await generate(
+                projectDir,
+                {
+                    ...config,
+                    deploy: {
+                        package: '$TMP'
+                    },
+                    options: {
+                        fioriTools: true,
+                        isRunningInBAS: true,
+                        isCloudProject: true
+                    },
+                    ui5: {
+                        ui5Version: '1.122.1'
+                    },
+                    customConfig: {
+                        adp: {
+                            safeMode: false
+                        }
+                    },
+                    flp: {
+                        bspName: 'bsp.test.app',
+                        languages: [
+                            {
+                                sap: 'testId',
+                                i18n: 'testKey'
+                            }
+                        ]
+                    }
+                },
+                fs
+            );
+            expect(
+                fs.dump(
+                    projectDir,
+                    (file) =>
+                        file.dirname === projectDir &&
+                        ['package.json', 'ui5.yaml', 'ui5-deploy.yaml'].includes(file.basename)
+                )
+            ).toMatchSnapshot();
+        });
+    });
+
+    describe('generateEnv', () => {
+        test('generate env file for s4hana cloud projects minimal', async () => {
+            const projectDir = join(outputDir, 'minimal');
+            await generateEnv(projectDir, '');
+            expect(
+                fs.dump(projectDir, (file) => file.dirname === projectDir && ['.env'].includes(file.basename))
+            ).toMatchSnapshot();
+        });
+
+        test('generate env file for s4hana cloud projects', async () => {
+            const projectDir = join(outputDir, 'env');
+            const data = `ABAP_USERNAME: test_name
+                          ABAP_PASSWORD: test_pass`;
+            await generateEnv(projectDir, data, fs);
+            expect(
+                fs.dump(projectDir, (file) => file.dirname === projectDir && ['.env'].includes(file.basename))
+            ).toMatchSnapshot();
+        });
+    });
+
+    describe('generateCf', () => {
```

Comment:
This can be removed as CF will not be part of this PR?

Reply:
Done